utils/client.go

```diff
@@ -2,40 +2,83 @@ package utils
 
 import (
 	"context"
+	"encoding/json"
 	"log"
 
 	"github.com/gorilla/websocket"
-	"github.com/satori/go.uuid"
+	uuid "github.com/satori/go.uuid"
 
 	"github.com/spaceuptech/space-cloud/model"
+	pb "github.com/spaceuptech/space-cloud/proto"
 )
 
 // Client is the object handling all client interactions
 type Client struct {
-	id      string
-	channel chan interface{}
-	Context context.Context
-	cancel  context.CancelFunc
-	socket  *websocket.Conn
+	id       string
+	channel  chan *model.Message
+	Context  context.Context
+	cancel   context.CancelFunc
+	socket   *websocket.Conn              //only for Websocket
+	stream   pb.SpaceCloud_RealTimeServer //Only for grpc
+	protocol RealTimeProtocol
 }
 
 // DataCallback is the callback invoked when data is read by the socket
 type DataCallback func(data *model.Message)
 
 // RoutineWrite starts a json writer routine
 func (c *Client) RoutineWrite() {
-	go func() {
-		for res := range c.channel {
-			err := c.socket.WriteJSON(res)
-			if err != nil {
-				log.Println(err)
+	switch c.protocol {
+	case Websocket:
+		go func() {
+			for res := range c.channel {
+				err := c.socket.WriteJSON(res)
+				if err != nil {
+					log.Println(err)
+				}
 			}
-		}
-	}()
+		}()
+	case GRPC:
+		go func() {
+			for res := range c.channel {
+				//Convert the Message into RealTime response.
+				switch res.Type {
+				case TypeRealtimeSubscribe, TypeRealtimeUnsubscribe:
+					//Decode the Message
+					feedData := make([]*pb.FeedData, 1)
```

Comment:
```suggestion
					feedData := make([]*pb.FeedData, len(responseMsg.Docs))
```

Reply:
you'll have to move line number 49 above 48

Reply:
How can I get the Length of the Docs?
Cos that will correspondingly be the Length of my array of pb.FeedData.

Reply:
Got it..Thanks!